packages/components/src/navigation/stories/index.js

```diff
@@ -1,88 +1,103 @@
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
 import Navigation from '../';
+import NavigationBackButton from '../back-button';
+import NavigationMenu from '../menu';
+import NavigationTitle from '../title';
 
 export default {
 	title: 'Components/Navigation',
 	component: Navigation,
 };
 
 const data = [
-	{ title: 'WooCommerce', slug: 'root', back: 'Dashboard' },
-	{ title: 'Home', slug: 'home', parent: 'root', menu: 'primary' },
+	{ title: 'Home', id: 'home' },
 	{
 		title: 'Analytics',
-		slug: 'analytics',
-		parent: 'root',
-		back: 'WooCommerce Home',
-		menu: 'primary',
+		id: 'analytics',
 	},
 	{
 		title: 'Orders',
-		slug: 'orders',
-		parent: 'root',
-		back: 'WooCommerce Home',
-		menu: 'primary',
+		id: 'orders',
 	},
 	{
 		title: 'Overview',
-		slug: 'overview',
+		id: 'overview',
 		parent: 'analytics',
 	},
 	{
 		title: 'Products report',
-		slug: 'products',
+		id: 'products',
 		parent: 'analytics',
 	},
 	{
 		title: 'All orders',
-		slug: 'all_orders',
+		id: 'all_orders',
 		parent: 'orders',
 	},
 	{
 		title: 'Payouts',
-		slug: 'payouts',
+		id: 'payouts',
 		parent: 'orders',
 	},
 	{
 		title: 'Settings',
-		slug: 'settings',
-		parent: 'root',
-		back: 'WooCommerce Home',
+		id: 'settings',
 		menu: 'secondary',
 	},
 	{
 		title: 'Extensions',
-		slug: 'extensions',
-		parent: 'root',
-		back: 'WooCommerce Home',
+		id: 'extensions',
 		menu: 'secondary',
 	},
 	{
 		title: 'General',
-		slug: 'general',
+		id: 'general',
 		parent: 'settings',
 	},
 	{
 		title: 'Tax',
-		slug: 'tax',
+		id: 'tax',
 		parent: 'settings',
 	},
 	{
 		title: 'My extensions',
-		slug: 'my_extensions',
+		id: 'my_extensions',
 		parent: 'extensions',
 	},
 	{
 		title: 'Marketplace',
-		slug: 'marketplace',
+		id: 'marketplace',
 		parent: 'extensions',
 	},
 ];
 
 function Example() {
-	return <Navigation data={ data } initial="home" />;
+	const [ active, setActive ] = useState( 'home' );
+	const items = data.map( ( item ) => {
+		item.onClick = () => setActive( item.id );
+		return item;
+	} );
+
+	return (
+		<Navigation active={ active } items={ items }>
+			<NavigationBackButton
+				nullText="Dashboard"
+				rootText="WooCommerce Home"
+				onClick={ ( item ) => ( item ? setActive( item.id ) : null ) }
+			/>
+			<NavigationTitle rootText="WooCommerce Home" />
```

Comment:
The back button is always supposed to go up just one level, so in a multi depth config this will need to be dynamic.

Reply:
I think the back button actually does more than go up one level.  According to the demo, it also sets the first item from the higher level as the active page.

The way it's set up here passes the item (from the parent) that will be activated.